File: Subprogramas/Sobrecarregados/Sobrecarregado.cs
//C Sharp

using System;

class Sobrecarregado {
    public void Imprimir(string s) {
        Console.WriteLine(s);
    }

    public void Imprimir(int i) {
        Console.WriteLine(i);
    }
}

class Program {
    static void Main() {
        Sobrecarregado p = new Sobrecarregado();
        p.Imprimir("Hello");  // chama o método que aceita string
        p.Imprimir(10);       // chama o método que aceita int
    }
}

File: Subprogramas/Sobrecarregados/Sobrecarregado.java
//Java

class Sobrecarregado {
    void imprimir(String s) {
        System.out.println(s);
    }

    void imprimir(int i) {
        System.out.println(i);
    }
}

public class Main {
    public static void main(String[] args) {
        Sobrecarregado p = new Sobrecarregado();
        p.imprimir("Hello");  // chama o método que aceita String
        p.imprimir(10);       // chama o método que aceita int
    }
}